infra/modules/database/role_manager/check.py

```diff
@@ -0,0 +1,53 @@
+import os
+
+from pg8000.native import Connection
+
+import db
+
+
+def check():
+    """Check that database roles, schema, and privileges were
+    properly configured
+    """
+    print("Running command 'check' to check database roles, schema, and privileges")
+    app_username = os.environ.get("APP_USER")
+    migrator_username = os.environ.get("MIGRATOR_USER")
+    schema_name = os.environ.get("DB_SCHEMA")
+
+    with (
+        db.connect_using_iam(app_username) as app_conn,
+        db.connect_using_iam(migrator_username) as migrator_conn,
+    ):
+        check_search_path(migrator_conn, schema_name)
+        check_migrator_create_table(migrator_conn, app_username)
+        check_app_use_table(app_conn)
+        cleanup_migrator_drop_table(migrator_conn)
+
+    return {"success": True}
+
+
+def check_search_path(migrator_conn: Connection, schema_name: str):
+    print("-- Check that search path is %s", schema_name)
+    assert db.execute(migrator_conn, "SHOW search_path") == [[schema_name]]
+
+
+def check_migrator_create_table(migrator_conn: Connection, app_username: str):
+    print(
+        "-- Check that migrator is able to create tables and grant access to app user: %s",
+        app_username,
+    )
+    cleanup_migrator_drop_table(migrator_conn)
+    db.execute(
+        migrator_conn, "CREATE TABLE IF NOT EXISTS temporary(created_at TIMESTAMP)"
+    )
+
```

Comment:
this is intentional. the first step of my testing is to remove this line to make sure the check fails on a new database without the DEFAULT PRIVILEGES (see step 2 in the platform-test PR)

<img width="814" alt="image" src="https://github.com/navapbc/template-infra/assets/447859/27aa1ce3-ca9b-4855-837b-5a9e4724e7cb">

if we include this line, then this check doesn't properly check the DEFAULT PRIVILEGES, since the check manually grants the app user access to the test table

Reply:
Ah I see. Then, we should probably change line 36 to: "-- Check that migrator is able to create tables"

Reply:
ah ok